sql: remove slow functions from TestRandomSyntaxFunctions

fqazi opened the pull request:
Previously, we could easily timeout running slower,
functions in the random syntax functions test. We
attempted to minimize this risk with resettable timeouts
which helped, but libpq has limited support for cancellation,
so we need to fully pull these out. To address this,
this patch will remove:
crdb_internal.revalidate_unique_constraints_in_all_tables
and crdb_internal.validate_ttl_scheduled_jobs from testing.

Fixes: #107929
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/108082)
<!-- Reviewable:end -->

fqazi committed:
sql: remove slow functions from TestRandomSyntaxFunctions

Previously, we could easily timeout running slower,
functions in the random syntax functions test. We
attempted to minimize this risk with resettable timeouts
which helped, but libpq has limited support for cancellation,
so we need to fully pull these out. To address this,
this patch will remove:
crdb_internal.revalidate_unique_constraints_in_all_tables
and crdb_internal.validate_ttl_scheduled_jobs from testing.

Fixes: #107929
Release note: None

chrisseto commented on pkg/sql/tests/rsg_test.go:
```diff
@@ -200,6 +200,14 @@ func (db *verifyFormatDB) execWithResettableTimeout(
 				}
 				return nil
+			case <-ctx.Done():
+				// Sanity: When our context gets cancelled and the test is
```
Are context cancellation and test timeout correlated? The current phrasing feels a bit ambiguous. Maybe something like "ctx is cancelled when the test is about to timeout. Log out whatever statement we're waiting on for debugging purposes. Sometimes queries won't respect cancellation due to lib/pq being weird". 

In the case that we do log out a query, are we trying to "save" the test from timing out or just leaving some breadcrumbs?

fqazi replied:
Updated the comment				
                                 // Sanity: The context is cancelled when the test is about to
				// timeout. We will log whatever statement we're waiting on for
				// debugging purposes. Sometimes queries won't respect
				// cancellation due to lib/pq limitations.

Yes,, the goal here is to get some bread crumbs out; otherwise, we have to look at server stacks to figure out what is executing. This gives us a better understanding modulo some noise in the successful cases.

chrisseto reviewed (approved):
LGTM! Just some semantic questions.

fqazi commented:
@chrisseto TFTR! 

bors r+
